In [1]:
import pandas as pd

df = pd.read_csv('../data/raw/WA_Fn-UseC_-HR-Employee-Attrition.csv')
df.shape

(1470, 35)

In [3]:
df.columns.tolist()

['Age',
 'Attrition',
 'BusinessTravel',
 'DailyRate',
 'Department',
 'DistanceFromHome',
 'Education',
 'EducationField',
 'EmployeeCount',
 'EmployeeNumber',
 'EnvironmentSatisfaction',
 'Gender',
 'HourlyRate',
 'JobInvolvement',
 'JobLevel',
 'JobRole',
 'JobSatisfaction',
 'MaritalStatus',
 'MonthlyIncome',
 'MonthlyRate',
 'NumCompaniesWorked',
 'Over18',
 'OverTime',
 'PercentSalaryHike',
 'PerformanceRating',
 'RelationshipSatisfaction',
 'StandardHours',
 'StockOptionLevel',
 'TotalWorkingYears',
 'TrainingTimesLastYear',
 'WorkLifeBalance',
 'YearsAtCompany',
 'YearsInCurrentRole',
 'YearsSinceLastPromotion',
 'YearsWithCurrManager']

Is Performance a Usable Factor?

Question: Are lower-rated employees more likely to leave?

Observation: The column contains only two values: 3 (1,244 employees, 84.6%) and 4 (226 employees, 15.4%). There are no ratings of 1 or 2.

Decision: This question was dropped. Comparing “good” with “very good” provides little operational value, and the absence of low performance ratings makes it impossible to draw any meaningful conclusions about the relationship between performance and turnover. This limitation should be noted in the analysis.

In [4]:
df['PerformanceRating'].value_counts()

PerformanceRating
3    1244
4     226
Name: count, dtype: int64

Overall Attrition Rate — Baseline Reference

Question: What proportion of employees left the company?

Observation: The overall attrition rate is 16.1% (237 out of 1,470 employees). The remaining 83.9% stayed with the company.

Decision: This rate becomes the baseline reference for the entire project. Every subgroup analyzed (department, overtime, tenure, etc.) will be compared against this 16.1% benchmark to determine whether it is overrepresented or underrepresented in employee attrition.

In [6]:
df['Attrition'].value_counts(normalize=True)

Attrition
No     0.838776
Yes    0.161224
Name: proportion, dtype: float64

Looking at the Attrition rate amongs the employees we can observe a 16,1% attrition rate